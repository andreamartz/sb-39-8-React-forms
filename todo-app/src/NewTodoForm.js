/**
 * NewTodoForm:
 *   * renders a form with one text input for the task to be created
 *   * controls the state of form data
 *   * when this form is submitted, a new Todo component should be created
 */

import React, { useState } from "react"

// const NewTodoForm = ({ addTodo }) => {
const NewTodoForm = ({ addTodo }) => {
  const INITIAL_STATE = {
    task: ''
  }

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData(data => ({       // data is the current form data
      ...data,
      [name]: value
    }));
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    addTodo({ ...formData });     // use the passed down fcn addTodo to create a new todo/task in state
    setFormData(INITIAL_STATE);   // reset the state of the form data
  }

  return (
    <div>
      <h1>Todo Form</h1>
      <form onSubmit={handleSubmit}>
        <label htmlFor="task">Todo</label>
        <input
          id="task"
          name="task"
          type="text"
          placeholder="task"
          value = {formData.task}
          onChange={handleChange}
        />
        <button>Add todo</button>
      </form>
    </div>
  )
}

export default NewTodoForm;