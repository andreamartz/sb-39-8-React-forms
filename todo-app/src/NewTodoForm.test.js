import React from "react";
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

describe("Smoke tests", () => {
  test('renders without crashing', () => {
    render(<NewTodoForm />);
  });
});

describe("Snapshot tests", () => {
  test('matches snapshot', () => {
    const {asFragment} = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});