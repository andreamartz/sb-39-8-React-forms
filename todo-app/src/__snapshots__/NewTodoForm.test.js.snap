// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Todo Form
    </h1>
    <form>
      <label
        for="task"
      >
        What task do you want to add?
      </label>
      <input
        id="task"
        name="task"
        placeholder="task"
        type="text"
        value=""
      />
      <button>
        Add todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
